src/harness/testcases/grant_testcase.py

```diff
@@ -624,3 +624,245 @@ def test_WINNF_FT_S_GRA_28(self):
+
+  @winnforum_testcase 
+  def test_WINFF_FT_S_GRA_6(self):
+    """No Authorized grant of CBSD .
+        Incumbent present in the PAL frequency range requested by the CBSD 
+        who is inside the protection zone identified by ESC. 
+    """
+
+    # Register cbsd
+    device_inside_esc_zone = json.load(
+        open(os.path.join('testcases', 'testdata', 'device_c.json')))
+    self._sas_admin.InjectFccId({'fccId': device_inside_esc_zone['fccId']})
+    request = {'registrationRequest': [device_inside_esc_zone]}
+    response = self._sas.Registration(request)['registrationResponse'][0]
+    # Check registration response
+    self.assertEqual(response['response']['responseCode'], 0)
+    cbsd_id = response['cbsdId']
+    del request, response
+    # Inject PAL Database Recoord
+    pal_database_record = json.load(
+        open(os.path.join('testcases', 'testdata', 'pal_database_record_0.json')))
+    pal_database_record['userId'] = device_inside_esc_zone['userId']
+    self._sas_admin.InjectPalDatabaseRecord(pal_database_record)
+    # Inject PPA Zone
+    ppa_zone = json.load(
+        open(os.path.join('testcases', 'testdata', 'ppa_zone_0.json')))
+    ppa_id = self._sas_admin.InjectZoneData(ppa_zone)
+    cluster_list = {'ppaId' : ppa_id['zoneId'], 'cbsdIds': [cbsd_id]}
+    self._sas_admin.InjectClusterList(cluster_list)
+    exclusion_zone = json.load(
+        open(os.path.join('testcases', 'testdata', 'exclusion_zone_0.json')))
+    zone_response = self._sas_admin.InjectEscZone(exclusion_zone)
+    trigger_esc_zone = {'zoneId': zone_response['zoneId'],
+                                    'frequencyRange': {
+                                     'lowFrequency': 3620000000.0,
+                                     'highFrequency': 3630000000.0}}
+    trigger_id = self._sas_admin.TriggerEscZone(trigger_esc_zone)
+    # grant request 
+    grant_0 = json.load(
+        open(os.path.join('testcases', 'testdata', 'grant_0.json')))
+    grant_0['cbsdId'] = cbsd_id
+    request = {'grantRequest': [grant_0]}
+    response = self._sas.Grant(request)['grantResponse'][0]
+    # Check grant response
+    self.assertEqual(response['cbsdId'], cbsd_id)
+    self.assertTrue(response['response']['responseCode'] in (0, 400))
+    if (response['response']['responseCode'] == 400):
+        self.assertFalse('grantId' in response)
+    else:
+        self.assertTrue('grantId' in response)
+        self.assertEqual(response['channelType'], 'PAL')
+        grant_id = response['grantId']
+        del request, response
+        request = {
+        'heartbeatRequest': [{
+            'cbsdId': cbsd_id,
+            'grantId': grant_id,
+            'operationState': 'GRANTED',
+            'grantRenew': True}]}
+        response = self._sas.Heartbeat(request)['heartbeatResponse'][0]
+        # Check the heartbeat response
+        grant_expire_time = datetime.strptime(response['grantExpireTime'], '%Y-%m-%dT%H:%M:%SZ')
```

Comment:
We will add import datetime, for the second point we think that HBT.3 is not correct and that the transmit expire time should be compared to the grant expire time of the same response.
in HBT.3:
- the grantExpire time in the grant response should be compared to grant expire time in the HB response.
- the transmit expire time should be compared to grant expire time in the HB response.
* because the SAS may change the grant expire time value between grant and HB response.
don't you agree?